TcgPocket/Features/Users/UsersController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using TcgPocket.Features.Roles;
+using TcgPocket.Features.Users.Commands;
+using TcgPocket.Features.Users.Dtos;
+using TcgPocket.Features.Users.Queries;
+using TcgPocket.Shared;
+
+namespace TcgPocket.Features.Users;
+
+[ApiController]
+[Route("/users")]
+public class UsersController : ControllerBase
+{
+    private readonly IMediator _mediator;
+
+    public UsersController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<Response<List<UserDto>>>> GetAllUsers()
+    {
+        var response = await _mediator.Send(new GetAllUsersQuery());
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpGet("{id:int}", Name = nameof(GetUserById))]
+    public async Task<ActionResult<Response<UserGetDto>>> GetUserById([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new GetUserByIdQuery { Id = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpGet("{id:int}/roles", Name = nameof(GetAllRolesByUserId))]
+    public async Task<ActionResult<Response<List<RoleGetDto>>>> GetAllRolesByUserId([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new GetAllRolesByUserIdQuery { UserId = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Response<UserGetDto>>> CreateUser([FromBody] UserCreateDto data)
+    {
+        var response = await _mediator.Send(new CreateUserCommand { User = data });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
+
+    [HttpPost("{id:int}/roles")]
+    public async Task<ActionResult<Response<UserRoleDto>>> AddRoleToUser([FromRoute] int id, [FromBody] AddRoleToUserCommand data)
+    {
+        var response = await _mediator.Send(new AddRoleToUserCommand { Id = id, RoleName = data.RoleName });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
```

Comment:
Should be getting just the `roleName` from the body, not the whole command

```suggestion
    [HttpPost("{id:int}/roles")]
    public async Task<ActionResult<Response<UserRoleDto>>> AddRoleToUser([FromRoute] int id, [FromBody] string roleName)
    {
        var response = await _mediator.Send(new AddRoleToUserCommand { Id = id, RoleName = roleName});

        return response.HasErrors
            ? BadRequest(response)
            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
    }
```